core/server/src/main/java/alluxio/worker/block/BlockLockManager.java

```diff
@@ -309,6 +328,48 @@ private void releaseBlockLockIfUnused(long blockId) {
   }
 
   /**
+   * Checks the internal state of the manager to make sure invariants hold.
+   *
+   * This method is intended for testing purposes. A runtime exception will be thrown if invalid
+   * state is encountered.
+   */
+  public void validate() {
```

Comment:
+1 to creating a coherent story on unit testing. The current practice in my experience has been to make judgement calls in these situations while erring on the side of not changing production code for testing. We currently do change production code in many places for the purposes of tests, e.g. `reset` methods, `HeartbeatContext#setTimerClass`, `MetricsConfig#getPropertyCategories`. I don't think there will ever be a hard and fast rule for this. In the case of this method, it doesn't expose any state on the object and could feasibly be used in actual production code, so I think it makes sense in `BlockLockManager`. Usually when we hide methods with reflection it's because they may accidentally be used to break abstraction.